### PR TITLE
HealthChecks.Dapr

### PR DESCRIPTION
**Pull request by bgelens**
**What this PR does / why we need it**:

Adds a HealthCheck for Dapr

**Which issue(s) this PR fixes**:

Health check for Dapr

Please reference the issue this PR will close: #712 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.Dapr/DependencyInjection/DaprHealthCheckBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public static IHealthChecksBuilder AddDapr(
+      this IHealthChecksBuilder builder,
+      string name = default,
+      HealthStatus? failureStatus = default,
+      IEnumerable<string> tags = default,
+      TimeSpan? timeout = default)
+    {
+      return builder.Add(new HealthCheckRegistration(
+        name ?? Name,
+        sp => new DaprHealthCheck(sp.GetRequiredService<DaprClient>()),
```
[the remark this answers is not in the document]
Comment (bgelens) [reply src/HealthChecks.Dapr/DependencyInjection/DaprHealthCheckBuilderExtensions.cs]: @sungam3r 

I noticed with RabbitMQ that it also uses a registration from DI container. Like with RabbitMQ, Dapr is also using a singleton registration. New DaprClients should not be created and the default one is already injected by [MVC Builder Extension](https://github.com/dapr/dotnet-sdk/blob/master/src/Dapr.AspNetCore/DaprMvcBuilderExtensions.cs)

I don't think using this in ConfigureServices method is very nice.

```csharp
services.AddHealthChecks()
        .AddDapr( daprClient: services.BuildServiceProvider().GetRequiredService<DaprClient>());
```

Could you give me a lead to how to solve this in a better way without using the DaprClient from DI Container. Or is it ok to keep using the code as is?

[the remark this answers is not in the document]
Comment (sungam3r) [reply src/HealthChecks.Dapr/DependencyInjection/DaprHealthCheckBuilderExtensions.cs]: I mean that you may provide **several** overloads for the caller to utilize different cases. Of course, calling `BuildServiceProvider` is a bad thing. You may provide overload with some `Action<DaprClientBuilder>` delegate to configure client and then inject created singleton.

> the default one is already injected by MVC Builder Extension

Good but only for those who uses MVC.

> Or is it ok to keep using the code as is?

I'm OK either way.

[the remark this answers is not in the document]
Comment (bgelens) [reply src/HealthChecks.Dapr/DependencyInjection/DaprHealthCheckBuilderExtensions.cs]: Thanks for getting back to me. I added an extra overload where someone can provide the DaprClient by argument. A DaprClient could be constructed like `DaprClientBuilder().Build();` Normally you would also add this to the DI container as singleton but now you could pass it in as well.

**Comment by sungam3r** [comment]
@bgelens Please rebase onto `master` if you are going to continue work on this PR.

**Comment by francesco-scullino** [comment]
Is this feature available in the nuget package?

**Comment by sungam3r** [comment]
PR was closed by @bgelens on 15 Mar. This code was not merged.

**Comment by sungam3r** [comment]
@cieciurm Would you like to revive this one?

**Comment by cieciurm** [comment]
Yeah, sure :)

cc @sungam3r

**Pull request closed without merging**